### PR TITLE
add variants of SeatCountBadge + minor layout adjustments

### PR DESCRIPTION
**Pull request by codingCookie**
- `SeatCountBadge`
  - add variant to place on a circle
  - add variant to place on an ellipse - with calculation of badge positioning
- `stories`: 
  - add stories for `SeatCountBadge`-variants

### CODE REVIEW

**Review by rieschl — APPROVED** [review]
The design is ok 👍

File: packages/seatmaps/src/SeatCountBadge.tsx [hunk 1]
```diff
@@ -1,22 +1,70 @@
 import { Badge } from './Badge';
 import { l } from './length';
+import { calculateBadgePosition } from './util/calculations';
 
 export interface SeatCountBadgeProps {
-    containerWidth: number;
```
Comment (MidnightDesign) [inline packages/seatmaps/src/SeatCountBadge.tsx:5]: That's a BC break.

File: packages/seatmaps/src/SeatCountBadge.tsx [hunk 2]
```diff
@@ -1,22 +1,70 @@
...
+    height: number;
+}
+
+// to place the badge on the right side of the volume (yes, that's on purpose, because unfortunately one value doesn't work for both shapes.)
 const HORIZONTAL_BADGE_PADDING = 2;
 const VERTICAL_BADGE_PADDING = 2;
+const HORIZONTAL_BADGE_PADDING_CIRCLE = 4;
+const VERTICAL_BADGE_PADDING_CIRCLE = 4;
+
+const badgePositionAdjustment = (containerWidth: number, isCircle = false) => {
```
Comment (MidnightDesign) [inline packages/seatmaps/src/SeatCountBadge.tsx]: Please don't use flag arguments.

File: packages/seatmaps/src/SeatCountBadge.tsx [hunk 3]
```diff
@@ -1,22 +1,70 @@
...
+            y={y}
+            color={color}
+            count={count}
+        />
+    );
+};
 
-export const SeatCountBadge = ({ containerWidth, count = 0, color = 'inherit' }: SeatCountBadgeProps) => {
-    const x = l(containerWidth) - HORIZONTAL_BADGE_PADDING;
+export const SeatCountBadgeOnCircle = ({ containerProps, count = 0, color = 'inherit' }: SeatCountBadgeProps) => {
```
Comment (MidnightDesign) [inline packages/seatmaps/src/SeatCountBadge.tsx]: That's a BC break.

File: packages/seatmaps/src/SeatCountBadge.tsx [hunk 4]
```diff
@@ -1,22 +1,70 @@
...
             x={x}
-            y={VERTICAL_BADGE_PADDING}
+            y={y}
+            color={color}
+            count={count}
+        />
+    );
+};
+
+export const SeatCountBadgeOnEllipse = ({ containerProps, count = 0, color = 'inherit' }: SeatCountBadgeProps) => {
```
Comment (MidnightDesign) [inline packages/seatmaps/src/SeatCountBadge.tsx]: That's a BC break.

File: packages/seatmaps/src/SeatCountBadge.tsx (hunk 2 above)
Comment (MidnightDesign) [reply packages/seatmaps/src/SeatCountBadge.tsx]: Also, please don't use nouns as function names.

**Review by MidnightDesign — CHANGES_REQUESTED** [review]

**Review by MidnightDesign — APPROVED** [review]

**Review by MidnightDesign — APPROVED** [review]

**Pull request merged**